src/browser_app_skeleton/src/components/shared/field.cr

```diff
@@ -1,34 +1,34 @@
-module Shared::Field
-  # This method is used to make it easier to render the same fields styles
-  # throughout your app
-  #
-  # ## Usage
-  #
-  #     field(form.email) { |i| email_input i }
-  #     field(form.email) { |i| email_input i, autofocus: "true" }
-  #
-  # ## Customization
-  #
-  # You can customize this method so that fields render like you expect
-  # For example, you might wrap it in a div with a "field-wrapper" class.
-  #
-  #    div class: "field-wrapper"
-  #      label_for field
-  #      yield field
-  #      errors_for fields
-  #    end
-  #
-  # You may also want to have more than one method if you render fields
-  # differently in different parts of your app, e.g. `compact_field`
-  private def field(field, hide_label : Bool = false, hide_errors : Bool = false, label_options = NamedTuple.new)
-    unless hide_label
-      label_for field, **label_options
-    end
+# This component is used to make it easier to render the same fields styles
+# throughout your app
+#
+# ## Usage
+#
+#     mount Shared::Field, form.email, &.email_input(autofocus: "true")
+#     mount Shared::Field, form.name, &.text_input
+#
+# ## Customization
+#
+# You can customize this class so that fields render like you expect
+# For example, you might wrap it in a div with a "field-wrapper" class.
+#
+#    div class: "field-wrapper"
+#      label_for field
+#      yield field
+#      errors_for fields
+#    end
+#
+# You may also want to have more more classes if you render fields
+# differently in different parts of your app, e.g. `Shared::CompactField``
+class Shared::Field(T) < BaseComponent
+  needs field : Avram::FillableField(T)
 
-    yield field
+  def render
+    label_for @field
 
-    unless hide_errors
-      errors_for field
+    with_defaults field: @field do |html|
+      yield html
     end
+
+    mount Shared::FieldErrors, @field
   end
```

Comment:
I still need to figure out a good way to hide labels and errors. I'm thinking maybe it can be another `needs` with a default, or I may change the `mount` method to use `.new` so you can call methods on the component before rendering: `mount Shared::Field.new(form.email).hide_label.hide_errors, &.email_input`

Reply:
I like the extensibility of this idea. If we’re moving towards objects over just methods, why not give the full power of objects?

Reply:
Yeah that’s kinda what I’m thinking to. Thanks for the feedback :) I think I’ll make that change in Lucky!